### PR TITLE
BCDA-3815: BFD Insights: Adding tables for BCDA insights queries

### PR DESCRIPTION
**Pull request by orenfromberg**
### Change Details

This branch adds glue catalog tables per environment for the postgres queries that we are currently making in BCDA using the insights data samplers.

These tables allow us to query any information that is being pushed through our firehoses.

#### Total number of data requests:
image

#### Current number of jobs in the queue:
image

Additionally, this fixes version of terraform to `0.13.4`.

### Acceptance Validation

- [x] We are able to analyze data from BCDA in AWS Athena.

### Feedback Requested

All feedback is welcome.

### External References

- [BCDA-3815](https://jira.cms.gov/browse/BCDA-3815)
- [BCDA-3873](https://jira.cms.gov/browse/BCDA-3873)

### Security Implications

- [ ] new software dependencies

- [ ] altered security controls

- [ ] new data stored or transmitted

- [ ] security checklist is completed for this change

- [ ] requires more information or team discussion to evaluate security implications

All feedback is welcome.

### CODE REVIEW

File: insights/terraform/projects/bcda/main.tf [hunk 1]
```diff
@@ -28,10 +28,68 @@ data "aws_kms_alias" "moderate_cmk" {
...
+}
+
+## Glue setup
+
+module "glue_jobs" {
+  source  = "../../modules/jobs"
+  project = local.project
+  tags    = local.tags
+
+  # Setup access to both the AB2D and common moderate bucket
```
Comment (msnook) [inline insights/terraform/projects/bcda/main.tf:58]: AB2D?   Is that a typo?

**Review by msnook — APPROVED** [review]
👍

**Review by sharshar-sb — APPROVED** [review]

**Comment by orenfromberg** [comment]
@jzulim I already applied the terraform but here it is in reverse:
```
------------------------------------------------------------------------

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  - destroy

Terraform will perform the following actions:

  # module.database.aws_glue_catalog_database.main will be destroyed
  - resource "aws_glue_catalog_database" "main" {
      - arn         = "arn:aws:glue:us-east-1:577373831711:database/bcda" -> null
      - catalog_id  = "577373831711" -> null
      - description = "BFD Insights database" -> null
      - id          = "577373831711:bcda" -> null
      - name        = "bcda" -> null
      - parameters  = {} -> null
    }

  # module.database.aws_glue_security_configuration.main will be destroyed
  - resource "aws_glue_security_configuration" "main" {
      - id   = "bcda" -> null
      - name = "bcda" -> null

      - encryption_configuration {
          - cloudwatch_encryption {
              - cloudwatch_encryption_mode = "DISABLED" -> null
            }

          - job_bookmarks_encryption {
              - job_bookmarks_encryption_mode = "DISABLED" -> null
            }

          - s3_encryption {
              - kms_key_arn        = "arn:aws:kms:us-east-1:577373831711:key/a4ca59d6-3978-434b-9a5c-c6ae8896db18" -> null
              - s3_encryption_mode = "SSE-KMS" -> null
            }
        }
    }

  # module.glue_jobs.aws_iam_policy.s3_access will be destroyed
  - resource "aws_iam_policy" "s3_access" {
      - arn    = "arn:aws:iam::577373831711:policy/bfd-insights/bfd-insights-bcda-glue-role-s3-access" -> null
      - id     = "arn:aws:iam::577373831711:policy/bfd-insights/bfd-insights-bcda-glue-role-s3-access" -> null
      - name   = "bfd-insights-bcda-glue-role-s3-access" -> null
      - path   = "/bfd-insights/" -> null
      - policy = jsonencode(
            {
              - Statement = [
                  - {
                      - Action   = [
                          - "s3:ListBucketMultipartUploads",
                          - "s3:ListBucket",
                          - "s3:HeadBucket",
                          - "s3:GetBucketLocation",
                        ]
                      - Effect   = "Allow"
                      - Resource = [
                          - "arn:aws:s3:::bfd-insights-moderate-577373831711",
                          - "arn:aws:s3:::bfd-insights-bcda-577373831711",
                        ]
                      - Sid      = "s3Buckets"
                    },
                  - {
                      - Action   = [
                          - "s3:PutObject*",
                          - "s3:ListMultipartUploadParts",
                          - "s3:GetObject*",
                          - "s3:DeleteObject*",
                          - "s3:AbortMultipartUpload",
                        ]
                      - Effect   = "Allow"
                      - Resource = [
                          - "arn:aws:s3:::bfd-insights-moderate-577373831711/*",
                          - "arn:aws:s3:::bfd-insights-bcda-577373831711/*",
                        ]
                      - Sid      = "s3Objects"
                    },
                  - {
                      - Action   = [
                          - "kms:ReEncrypt*",
                          - "kms:GenerateDataKey*",
                          - "kms:Encrypt",
                          - "kms:DescribeKey",
                          - "kms:Decrypt",
                        ]
                      - Effect   = "Allow"
                      - Resource = [
                          - "arn:aws:kms:us-east-1:577373831711:key/a4ca59d6-3978-434b-9a5c-c6ae8896db18",
                          - "arn:aws:kms:us-east-1:577373831711:alias/bfd-insights-moderate-cmk",
                        ]
                      - Sid      = "CMK"
                    },
                ]
              - Version   = "2012-10-17"
            }
        ) -> null
    }

  # module.glue_jobs.aws_iam_role.glue_role will be destroyed
  - resource "aws_iam_role" "glue_role" {
      - arn                   = "arn:aws:iam::577373831711:role/bfd-insights/bfd-insights-bcda-glue-role" -> null
      - assume_role_policy    = jsonencode(
            {
              - Statement = [
                  - {
                      - Action    = "sts:AssumeRole"
                      - Effect    = "Allow"
                      - Principal = {
                          - Service = "glue.amazonaws.com"
                        }
                      - Sid       = "assume"
                    },
                ]
              - Version   = "2012-10-17"
            }
        ) -> null
      - create_date           = "2020-12-30T16:10:40Z" -> null
      - description           = "Allow the Glue service to access the Insights buckets" -> null
      - force_detach_policies = false -> null
      - id                    = "bfd-insights-bcda-glue-role" -> null
      - max_session_duration  = 3600 -> null
      - name                  = "bfd-insights-bcda-glue-role" -> null
      - path                  = "/bfd-insights/" -> null
      - tags                  = {
          - "application" = "bfd-insights"
          - "business"    = "OEDA"
          - "project"     = "bcda"
        } -> null
      - unique_id             = "AROAYM3RJQIP33VC6FWRK" -> null
    }

  # module.glue_jobs.aws_iam_role_policy_attachment.glue_role_attach[0] will be destroyed
  - resource "aws_iam_role_policy_attachment" "glue_role_attach" {
      - id         = "bfd-insights-bcda-glue-role-20201230161041281600000001" -> null
      - policy_arn = "arn:aws:iam::577373831711:policy/bfd-insights/bfd-insights-bcda-glue-role-s3-access" -> null
      - role       = "bfd-insights-bcda-glue-role" -> null
    }

  # module.glue_jobs.aws_iam_role_policy_attachment.glue_role_attach[1] will be destroyed
  - resource "aws_iam_role_policy_attachment" "glue_role_attach" {
      - id         = "bfd-insights-bcda-glue-role-20201230161041294500000003" -> null
      - policy_arn = "arn:aws:iam::577373831711:policy/bfd-insights/bfd-insights-bcda-glue-role-s3-access" -> null
      - role       = "bfd-insights-bcda-glue-role" -> null
    }

  # module.glue_jobs.aws_iam_role_policy_attachment.glue_role_attach[2] will be destroyed
  - resource "aws_iam_role_policy_attachment" "glue_role_attach" {
      - id         = "bfd-insights-bcda-glue-role-20201230161041293000000002" -> null
      - policy_arn = "arn:aws:iam::577373831711:policy/bfd-insights/bfd-insights-bcda-glue-role-s3-access" -> null
      - role       = "bfd-insights-bcda-glue-role" -> null
    }

  # module.insights_data_table_opensbx.aws_glue_catalog_table.aws_glue_catalog_table will be destroyed
  - resource "aws_glue_catalog_table" "aws_glue_catalog_table" {
      - arn           = "arn:aws:glue:us-east-1:577373831711:table/bcda/opensbx_insights" -> null
      - catalog_id    = "577373831711" -> null
      - database_name = "bcda" -> null
      - description   = "opensbx insights data" -> null
      - id            = "577373831711:bcda:opensbx_insights" -> null
      - name          = "opensbx_insights" -> null
      - parameters    = {
          - "EXTERNAL" = "TRUE"
        } -> null
      - retention     = 0 -> null
      - table_type    = "EXTERNAL_TABLE" -> null

      - partition_keys {
          - comment = "Approximate delivery time" -> null
          - name    = "dt" -> null
          - type    = "string" -> null
        }

      - storage_descriptor {
          - bucket_columns            = [] -> null
          - compressed                = false -> null
          - input_format              = "org.apache.hadoop.mapred.TextInputFormat" -> null
          - location                  = "s3://bfd-insights-bcda-577373831711/databases/bcda/opensbx_insights" -> null
          - number_of_buckets         = 0 -> null
          - output_format             = "org.apache.hadoop.hive.ql.io.HiveIgnoreKeyTextOutputFormat" -> null
          - parameters                = {} -> null
          - stored_as_sub_directories = false -> null

          - columns {
              - name = "name" -> null
              - type = "string" -> null
            }
          - columns {
              - name = "timestamp" -> null
              - type = "bigint" -> null
            }
          - columns {
              - name = "json_result" -> null
              - type = "string" -> null
            }

          - ser_de_info {
              - name                  = "opensbx_insights" -> null
              - parameters            = {
                  - "dots.in.keys"          = "true"
                  - "ignore.malformed.json" = "true"
                  - "timestamp.formats"     = "yyyy-MM-dd'T'HH:mm:ss.SSSSSS,yyyy-MM-dd'T'HH:mm:ss.SSS,yyyy-MM-dd'T'HH:mm:ss,yyyy-MM-dd'T'HH:mm:ss.SSSSSS'Z',yyyy-MM-dd'T'HH:mm:ss.SSS'Z',yyyy-MM-dd'T'HH:mm:ss'Z'"
                } -> null
              - serialization_library = "org.apache.hive.hcatalog.data.JsonSerDe" -> null
            }
        }
    }

  # module.insights_data_table_prod.aws_glue_catalog_table.aws_glue_catalog_table will be destroyed
  - resource "aws_glue_catalog_table" "aws_glue_catalog_table" {
      - arn           = "arn:aws:glue:us-east-1:577373831711:table/bcda/prod_insights" -> null
      - catalog_id    = "577373831711" -> null
      - database_name = "bcda" -> null
      - description   = "prod insights data" -> null
      - id            = "577373831711:bcda:prod_insights" -> null
      - name          = "prod_insights" -> null
      - parameters    = {
          - "EXTERNAL" = "TRUE"
        } -> null
      - retention     = 0 -> null
      - table_type    = "EXTERNAL_TABLE" -> null

      - partition_keys {
          - comment = "Approximate delivery time" -> null
          - name    = "dt" -> null
          - type    = "string" -> null
        }

      - storage_descriptor {
          - bucket_columns            = [] -> null
          - compressed                = false -> null
          - input_format              = "org.apache.hadoop.mapred.TextInputFormat" -> null
          - location                  = "s3://bfd-insights-bcda-577373831711/databases/bcda/prod_insights" -> null
          - number_of_buckets         = 0 -> null
          - output_format             = "org.apache.hadoop.hive.ql.io.HiveIgnoreKeyTextOutputFormat" -> null
          - parameters                = {} -> null
          - stored_as_sub_directories = false -> null

          - columns {
              - name = "name" -> null
              - type = "string" -> null
            }
          - columns {
              - name = "timestamp" -> null
              - type = "bigint" -> null
            }
          - columns {
              - name = "json_result" -> null
              - type = "string" -> null
            }

          - ser_de_info {
              - name                  = "prod_insights" -> null
              - parameters            = {
                  - "dots.in.keys"          = "true"
                  - "ignore.malformed.json" = "true"
                  - "timestamp.formats"     = "yyyy-MM-dd'T'HH:mm:ss.SSSSSS,yyyy-MM-dd'T'HH:mm:ss.SSS,yyyy-MM-dd'T'HH:mm:ss,yyyy-MM-dd'T'HH:mm:ss.SSSSSS'Z',yyyy-MM-dd'T'HH:mm:ss.SSS'Z',yyyy-MM-dd'T'HH:mm:ss'Z'"
                } -> null
              - serialization_library = "org.apache.hive.hcatalog.data.JsonSerDe" -> null
            }
        }
    }

Plan: 0 to add, 0 to change, 9 to destroy.

------------------------------------------------------------------------
```

**Review by jzulim — APPROVED** [review]
LGTM

**Pull request merged**